greymatter-io/gitops-core
# OpenShift support for Prometheus and Vector

[sc-27774]


This PR includes fixes for Prometheus and Vector in OpenShift, by creating and binding SCCs to permission them appropriately.

I also fixed another regression I found in the OpenShift support which only occurs when `spire` is `true`.

I also pointed to Redis and Prometheus images from Quay in place of Docker Hub to avoid their rate limiting. (On that point, Vector is the only image we don't pull from Quay, and I didn't change it because there wasn't an official one.)

[sc-28198]

I also replaced `+` with `list.Concat` in appropriate places in EXTRACTME.cue files to improve performance in CUE loading.

I also fixed a regression in gitops-core that was causing the observables app to look for manually-configured certificates even if you hadn't configured it for that.

## How to test

I have already run the operator integration tests against this branch, and caught some regressions doing that. They're passing now: https://buildkite.com/greymatter/operator/builds/1373

To test that this works in OpenShift, you'll need to deploy them into an OpenShift cluster with `config.openshift` set to `true`. Try with and without `spire`. I have one up, so it's probably easiest to pair on this, but if you have better luck than I did perhaps you'll be able to follow these instructions to get yourself a cluster: https://wiki.greymatter.io/en/internal-infra/runbooks/deploy-openshift-cluster

0. Get an OpenShift cluster running and authenticate to it.
1. From this branch, deploy into OpenShift with the GitOps target pointing where you definitely certainly have `config.openshift` on, and then deploy the operator from a `cue eval` and _not_ from operator.yaml or operator_withspire.yaml. Neither of those have the necessary operator-install-time security adjustments necessary for OpenShift. You want one of these instead:
```
(
cd pkg/cuemodule/core
kubectl create namespace gm-operator

kubectl create secret docker-registry gm-docker-secret \
  --docker-server=quay.io \
  --docker-username=$QUAY_USERNAME \
  --docker-password=$QUAY_PASSWORD \
  -n gm-operator

kubectl create secret generic greymatter-sync-secret \
  --from-file=ssh-private-key=$HOME/.ssh/id_ed25519 \
  -n gm-operator

cue eval -c ./k8s/outputs --out text \
         -t operator_image=quay.io/greymatterio/operator:0.11.0 \
         -e operator_manifests_yaml | kubectl apply -f -
)
```
2. If everything comes up (except for the observables pipeline service, because you didn't configure ElasticSearch) then it worked. In particular, do you see Prometheus and Vector ("greymatter-audit-agent") pods in the `greymatter` namespace? If the pods are there at all (again, the Vector pods won't actually be working unless you configured ElasticSearch) then this PR is successful.
